src/main/java/listeners/MainEventListener.java

```diff
@@ -0,0 +1,232 @@
+package listeners;
+
+import entities.Server;
+import entities.Topic;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Activity;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.ReadyEvent;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class MainEventListener extends ListenerAdapter {
+    /**
+     * A function that handles a single command.
+     */
+    private interface CommandHandler {
+        /**
+         * Handle a single command.
+         * @param member The member that called the command
+         * @param channel The channel that the command was called in
+         * @param server The Server that the command was called in
+         * @param args Extra command arguments, if any
+         */
+        void handle(@NotNull Member member, TextChannel channel, Server server, String[] args);
+    }
+
+    private final HashMap<String, Server> servers = new HashMap<>();
+
+    /**
+     * Check if the given Member has administrator permissions. Notifies the user if they do not have permission.
+     * @param member The Member to check
+     * @return True if the Member has admin, false otherwise
+     */
+    private static boolean checkAdmin(Member member, TextChannel channel) {
+        if (!member.hasPermission(Permission.ADMINISTRATOR)) {
+            channel.sendMessage(member.getAsMention() + " You must have administrator permission to run this command.").queue();
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Check if the given Topic exists in the Server. Notifies the user if the topic does not exist.
+     * @param member The Member to reply to
+     * @param channel The Channel to reply within
+     * @param server The Server to check in
+     * @param topic The Topic to check for
+     * @return Contains the Topic, if it exists
+     */
+    private static Optional<Topic> checkTopicExists(Member member, TextChannel channel, Server server, String topic) {
+        Optional<Topic> topicOptional = Arrays.stream(server.getTopics())
+                .filter(t -> t.getName().equalsIgnoreCase(topic))
+                .findFirst();
+        if (topicOptional.isEmpty()) {
+            channel.sendMessage(String.format(
```

Comment:
I'd definitely recommend refactoring error messages into a static class -- see [here](https://github.com/codeRIT/brickhack-mobile-ios/blob/master/BrickHack-Mobile/AlertMessage.swift) for iOS, although this seems much easier to present given it's text based 😅

Reply:
What benefits would this have? I'm not saying it's a bad idea, just that I don't fully understand why that would be better.

Reply:
Never mind; I just saw [this comment](https://github.com/codeRIT/mentorbot/pull/3/files/f6d09b4f9f565d44adc98382deaaa87b345c53a8#r515673643) and it makes sense now.

Reply:
Also included in #9.